### PR TITLE
upgrade to Symfony 6.3

### PR DESCRIPTION
**Pull request by hbagherzadegan**
### please **don't merge this** until all service ready for symfony 6.3

### CODE REVIEW

**Comment by hbagherzadegan** [comment]
we have some specific methods which use in behat-context package which is not exist in macpaw/behat-messenger-context , to remove this package we should adjust and remove using that method in all package which use BehatMessengerContext check this : https://kollex.atlassian.net/browse/DEV-4047

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -14,17 +14,91 @@ jobs:
                 php:
                     - '7.4'
                     - '8.0'
+                    - '8.1'
```
Comment (JarJak) [inline .github/workflows/ci.yaml]: we don't need any version other than php 8.2 and symfony 6.3

File: composer.json [hunk 2]
```diff
@@ -26,16 +26,19 @@
     ],
     "license": "MIT",
     "require": {
-        "php": "^7.4 || ^8.0",
-        "behat/behat": "^3.0",
-        "symfony/messenger": "^4.4 || ^5.0",
-        "symfony/serializer": "^4.4 || ^5.0"
+        "php": ">=8.1",
```
Comment (JarJak) [inline composer.json]: 8.2

File: rector.php [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+declare(strict_types=1);
+
+use Rector\Config\RectorConfig;
+use Rector\Symfony\Set\SymfonyLevelSetList;
+use Rector\Symfony\Set\SymfonySetList;
+
+return static function (RectorConfig $rectorConfig): void {
+    $rectorConfig->sets(
+        [
+            SymfonyLevelSetList::UP_TO_SYMFONY_63,
```
Comment (JarJak) [inline rector.php:22]: can we upgrade php as well?

File: src/Context/MessengerContext.php [hunk 4]
```diff
@@ -154,10 +164,11 @@ public function allTransportMessagesShouldBeJsonWithVariableFields(
 
     /**
      * @Then there is :expectationMessageCount messages in transport :transportName
+     * @throws Exception
      */
     public function thereIsCountMessagesInTransport(int $expectedMessageCount, string $transportName): void
     {
-        $actualMessageCount = count($this->getEnvelopesFromTransport($transportName));
+        $actualMessageCount = count(new Collection($this->getEnvelopesFromTransport($transportName)));
```
Comment (JarJak) [inline src/Context/MessengerContext.php]: why this needs to be collected before count?

**Review by JarJak — CHANGES_REQUESTED** [review]

**Pull request closed without merging**